Mathlib/Combinatorics/Enumerative/DyckWord.lean

```diff
@@ -142,17 +150,21 @@ def nest : DyckWord where
     rw [add_comm]
     exact add_le_add (zero_le _) ((count_le_length _ _).trans (by simp))
 
+lemma nest_add_ne_empty : (p.nest + q) ≠ 0 := by rw [← toList_ne_nil]; change U :: _ ++ _ ≠ []; simp
```

Comment:
Can you have a simp lemma of the form `p + q = 0 ↔ p = 0 ∧ q = 0` and a non-simp lemma of the form `p + q ≠ 0 ↔ p ≠ 0 ∨ q ≠ 0`? Then you wouldn't need `nest_add_ne_empty`
```suggestion
```